hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -603,7 +602,8 @@ observeInitTx ::
   [Party] ->
   Tx ->
   Either NotAnInitReason InitObservation
-observeInitTx networkId cardanoKeys expectedCP party allConfiguredParties tx = do
+observeInitTx networkId cardanoKeys expectedCP party otherParties tx = do
+  let allConfiguredParties = party : otherParties
```

Comment:
Could: put this in a where clause.

I personally like the top-down approach better and also proposed it as a [coding standard ](https://github.com/input-output-hk/hydra/wiki/Coding-Standards#proposal-function-structure)